kinto/core/storage/testing.py

```diff
@@ -1525,6 +1538,56 @@ def test_get_all_handle_all_pagination_rules(self):
         self.assertEqual(total_records, 10)
         self.assertEqual(len(records), 4)
 
+    def test_get_all_parent_id_paginates_correctly(self):
+        """Verify that pagination doesn't squash or duplicate some records"""
+
+        for parent in range(10):
+            parent_id = 'abc{}'.format(parent)
+            self.storage.create(parent_id=parent_id, collection_id='c',
+                                record={'id': 'some_id', 'secret_data': parent_id})
+
+        real_records, _ = self.storage.get_all(parent_id='abc*', collection_id='c')
+        self.assertEqual(len(real_records), 10)
+
+        def sort_by_secret_data(l):
+            return sorted(l, key=lambda r: r['secret_data'])
+
+        GT = utils.COMPARISON.GT
+        LT = utils.COMPARISON.LT
+        for order in [('secret_data', 1), ('secret_data', -1)]:
+            order_field, order_direction = order
+            sort = [Sort(*order), Sort('last_modified', -1)]
+            pagination_direction = GT if order_direction == 1 else LT
```

Comment:
I know it makes no sense to redo that at each loop iteration, but I would have had less trouble reading all this if `order_field` and `pagination_direction` would have been define right along the pagination rules below :)